File: packages/expo-dev-menu/vendored/react-native-safe-area-context/android/devmenu/com/th3rdwave/safeareacontext/InsetsChangeEvent.java
package devmenu.com.th3rdwave.safeareacontext;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.events.Event;
import com.facebook.react.uimanager.events.RCTEventEmitter;

/* package */  class InsetsChangeEvent extends Event<InsetsChangeEvent> {
  static final String EVENT_NAME = "topInsetsChange";

  private EdgeInsets mInsets;
  private Rect mFrame;

  InsetsChangeEvent(int viewTag, EdgeInsets insets, Rect frame) {
    super(viewTag);

    mInsets = insets;
    mFrame = frame;
  }

  @Override
  public String getEventName() {
    return EVENT_NAME;
  }

  @Override
  public void dispatch(RCTEventEmitter rctEventEmitter) {
    WritableMap event = Arguments.createMap();
    event.putMap("insets", SerializationUtils.edgeInsetsToJsMap(mInsets));
    event.putMap("frame", SerializationUtils.rectToJsMap(mFrame));
    rctEventEmitter.receiveEvent(getViewTag(), getEventName(), event);
  }
}

File: packages/expo-dev-menu/vendored/react-native-safe-area-context/android/devmenu/com/th3rdwave/safeareacontext/InsetsChangeEvent.kt
package devmenu.com.th3rdwave.safeareacontext

import com.facebook.react.uimanager.PixelUtil
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class InsetsChangeEvent(
  insets: EdgeInsets,
  frame: Rect
) : Record {
  @Field("insets")
  val parsedInsets = EdgeInsets(
    PixelUtil.toDIPFromPixel(insets.top),
    PixelUtil.toDIPFromPixel(insets.right),
    PixelUtil.toDIPFromPixel(insets.bottom),
    PixelUtil.toDIPFromPixel(insets.left)
  )

  @Field("frame")
  val parsedFrame = Rect(
    PixelUtil.toDIPFromPixel(frame.x),
    PixelUtil.toDIPFromPixel(frame.y),
    PixelUtil.toDIPFromPixel(frame.width),
    PixelUtil.toDIPFromPixel(frame.height)
  )
}
